### PR TITLE
Fix ARM CI build.

### LINKED ISSUE #144: "Linux ARM Release" CI broken
This leg has consistently failed very quickly: https://ci.dot.net/job/dotnet_source-build/job/master/job/Linux%20ARM_Release_prtest/

For example:

```
16:30:58 + ./init-tools.sh
16:30:58 cat: /mnt/j/workspace/dotnet_source-build/master/Linux: No such file or directory
16:30:58 cat: ARM_Release_prtest/BuildToolsVersion.txt: No such file or directory
16:30:58 cat: /mnt/j/workspace/dotnet_source-build/master/Linux: No such file or directory
16:30:58 cat: ARM_Release_prtest/DotnetCLIVersion.txt: No such file or directory
16:30:58 ./init-tools.sh: line 46: [: /mnt/j/workspace/dotnet_source-build/master/Linux: binary operator expected
16:30:58 Tools are already initialized
16:30:58 [Linux ARM_Release_prtest] $ /bin/sh -xe /tmp/hudson654491549140821879.sh
16:30:58 + ./arm-ci.sh arm ./build.sh /p:Platform=arm /p:Configuration=Release /p:IsJenkinsBuild=true
16:30:58 docker: invalid reference format.
16:30:58 See 'docker run --help'.
16:30:58 Build step 'Execute shell' marked build as failure
```

@crummel Should this leg be removed, or is it expected to succeed?

### PR DESCRIPTION
**Pull request by crummel**
Fixes #144.
- Added arm-build script to do some extra needed setup on ARM.
- Removed unneeded -RuntimeOS arguments.
- Made sure we pass platform through wherever needed.
- Made Docker run with more capabilities (CoreCLR's ARM build does this as it needs access to the kernel).
- Removed duplicate arguments in ARM build.
- Added a few more dummy packages.

### CODE REVIEW

**Comment by crummel** [comment]
@dotnet-bot test ci please

**Comment by crummel** [comment]
I expect this to work with updated CI - I actually messed up a merge and brought some RID stuff back in after Eric fixed it.

**Comment by eerhardt** [comment]
WHOO HOO!  look at the ARM build pass!

File: arm-build.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+set -euo pipefail
```
Comment (weshaggard) [inline arm-build.sh]: Do we really need another script verse just adding some conditions for arm in the existing scripts?

File: arm-build.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
```
Comment (eerhardt) [inline arm-build.sh]: Is there any way we can share this logic with `build.sh`?  There seems to be a fair amount of duplication.

File: arm-ci.sh [hunk 3]
```diff
@@ -12,6 +12,6 @@ elif [ "$__buildArch" == "armel" ]; then
     __dockerImage=" hqueue/dotnetcore:ubuntu1404_cross_prereqs_v4-tizen_rootfs"
     __dockerEnvironmentVariables+=" -e ROOTFS_DIR=/crossrootfs/armel.tizen.build"
 fi
-__dockerCmd="sudo docker run ${__dockerEnvironmentVariables} --privileged -i --rm -v $__currentWorkingDirectory:/opt/code -w /opt/code $__dockerImage"
+__dockerCmd="sudo docker run ${__dockerEnvironmentVariables} --privileged --cap-add=ALL -v /dev:/dev -v /lib/modules:/lib/modules -i --rm -v $__currentWorkingDirectory:/opt/code -w /opt/code $__dockerImage"
```
Comment (eerhardt) [inline arm-ci.sh]: what is `/dev` needed for?

**Comment by eerhardt** [comment]
I assume we will tackle the Tizen CI build in a separate PR?

**Comment by crummel** [comment]
Had some differences between my local build and the CI build, so this iteration might not pass, but if so it gets rid of `arm-build.sh` and passing through `/dev`.
@dotnet-bot test ci please

File: arm-build.sh (hunk 1 above)
Comment (crummel) [reply arm-build.sh]: Removed, this was due to difference between my local machine and CI.

File: arm-build.sh (hunk 2 above)
Comment (crummel) [reply arm-build.sh]: Removed, this was due to difference between my local machine and CI.

File: arm-ci.sh (hunk 3 above)
Comment (crummel) [reply arm-ci.sh]: Removed, this was due to difference between my local machine and CI.

**Comment by crummel** [comment]
@eerhardt @weshaggard Please take a look again, I think I've taken care of the issues.  Tizen will come in another PR, hopefully soon - I'm looking at it now.

**Pull request merged**